src/Altinn.AccessManagement.Core/Services/AuthorizedPartiesService.cs

```diff
@@ -285,6 +310,36 @@ private async Task<List<AuthorizedParty>> BuildAuthorizedParties(int subjectUser
             authorizedParty.EnrichWithResourceAccess(delegation.ResourceId);
         }
 
+        var instanceDelegations = await GetInstanceDelegations(subjectUserId, subjectPartyIds, cancellationToken);
+        var instanceParties = await _contextRetrievalService.GetPartiesByUuids(instanceDelegations.Select(i => i.FromUuid), false, cancellationToken);
+        foreach (var delegation in instanceDelegations)
+        {
+            if (!instanceParties.TryGetValue(delegation.FromUuid.ToString(), out var instanceParty))
+            {
+                throw new UnreachableException($"Get AuthorizedParties failed to lookup party with uuid {delegation.FromUuid} while building instance delegations list");
+            }
+
+            // Found existing party. Need to just append instances 
+            if (!authorizedPartyDict.TryGetValue(instanceParty.PartyId, out var authorizedParty))
+            {
+                authorizedParty = new AuthorizedParty(instanceParty);
+                authorizedPartyDict.Add(authorizedParty.PartyId, authorizedParty);
```

Comment:
Mangler logikk for dersom AuthorizedParty på instansdelegeringen er en underenhet. Hvor da hovedenhet skal legges til med "onlyHierarchyElementWithNoAccess": true, og underenheten som SubUnit (AC-3/-8/-9)

Kan se på det